include/GoogleSync/GoogleSync.php

```diff
@@ -1143,25 +311,43 @@ public function syncAllUsers()
     {
 
         // First we populate the array of syncable users
-        $this->setSyncUsers();
+        try {
+            $ret = $this->setSyncUsers();
+        } catch (Exception $e) {
+            // TODO: handle the exception instead just loggin it, it will be logged if unhandled by suitecrm core.
```

Comment:
@gymad 
I'm not sure what do to here. I don't want the syncAllUsers() to stop here if a single user has a configuration issue that throws an exception. It should move on to the next user. After it's all done, if there were any errors, it will return false to the cron runner so it's marked as unsuccessful.
Should there be some kind of user notification given if it fails? Maybe send the user a email, or some kind of popup on their screen. I don't know. At this point I'm not sure increasing the scope of the project is advisable. I'm already crammed for time. I really need to get this PR pulled so the fixes are in the next RC.

Reply:
I'll just let SuiteCRM log the exception for now.

Reply:
Err... I can't do that. It stops the foreach loop from progressing to the next loop. I'm catching it here, and logging it. It's all I can do with it at this point. I'm logging a simple 'exception caught' message to fatal, and a stack trace to error.